fpm/test/new_test/new_test.f90

```diff
@@ -0,0 +1,286 @@
+         case default
+            write(*,*)'ERROR: internal error. unknown directory name ',trim(directories(i))
+            stop 4
+         end select
+         !! MSwindows has hidden files in it
+         call list_files(trim(directories(i)), file_names,recurse=.true.)
+         if(allocated(fnames))deallocate(fnames)
+         allocate(character(len=0) :: fnames(0))
+         do j=1,size(file_names)
+           if(file_names(j)%s(1:1).eq.'.'.or.index(file_names(j)%s,'/.').ne.0.or.index(file_names(j)%s,'\.').ne.0)cycle
+           fnames=[character(len=max(len(fnames),len(file_names(j)%s))) :: fnames,file_names(j)%s]
+         enddo
+         write(*,'(*(g0))',advance='no')'>>>DIRECTORY ',trim(directories(i)),':   '
+         write(*,'(*(g0:,", "))')( file_names(j)%s, j=1,size(file_names) )
+         if(size(expected).ne.size(fnames))then
+            write(*,*)'unexpected number of files in file list=',size(fnames),' expected ',size(expected)
+            tally=[tally,.false.]
+            cycle TESTS
+         else
+            select case (get_os_type()) 
+            case (OS_UNKNOWN, OS_LINUX, OS_MACOS, OS_CYGWIN, OS_SOLARIS, OS_FREEBSD)
+            case (OS_WINDOWS) 
+               do j=1,size(expected)
+                  expected(j)=u2d(expected(j))
+               enddo 
+            case default
+               write(*,*)'ERROR: unknown OS. Stopping test'
+               stop 3
+            end select 
+            do j=1,size(expected)
+               if( .not.any(fnames(j)==expected) )then
+                  tally=[tally,.false.]
+                  write(*,'("ERROR: EXPECTED ",*(g0:,", "))')( trim(expected(k)), k=1,size(expected) )
+                  write(*,'(*(g0))')'       NO MATCH FOR ',fnames(j)
+                  cycle TESTS
+               endif
+            enddo
+            tally=[tally,.true.]
+         endif
+      endif
+   enddo TESTS
+   write(*,'("TALLY=",*(g0))')tally
+   if(all(tally))then
+      write(*,'(*(g0))')'PASSED: all ',count(tally),' tests passed '
+   else
+      write(*,*)'FAILED: PASSED=',count(tally),' FAILED=',count(.not.tally)
+      stop 5
+   endif
+!-----------------------------------------------------------------------------------------------------------------------------------
+contains
+!-----------------------------------------------------------------------------------------------------------------------------------
+function u2d(pathname) result(dos)
+! simplistically replace / with \ to make posix pathname DOS pathname
+character(len=*),intent(in) :: pathname
+character(len=:),allocatable :: dos
+integer :: i
+dos=pathname
+do i=1,len(pathname)
+   if(pathname(i:i).eq.'/')dos(i:i)='\'
+enddo
+end function u2d
+!-----------------------------------------------------------------------------------------------------------------------------------
+function djb2_hash_arr(chars,continue) result(hash_128)
```

Comment:
This seems to be unused?

Reply:
djb2_hash_arr is unused at this point. I was considering whether the output files generated were stable enough to do a basic hash sum on them as a check for expected output and a few other tests but this all seemed stalled and there were a few other issues that developed from making the test such as list_files() not having the "-a" switch on the ls(1) command but showing hidden files on MSWindows side;  mkdir failing on an existing directory on the MSWindows side but silently continuing on the POSIX side that should be very short separate PRs, etc. So the new_test.f90 was sufficient to show the commands worked in the automated tests on the github site and were passing on all platforms so I was leaving it at that for now. Assuming the stdlib will at one point add routines such as in my module P_process and some higher-level checksum functions (which are tools I use to verify executables with) I thought I would leave well enough alone, especially since this all seemed stalled just when a fully functional prototype seemed a few days away.  So that was in there for "future expansion" and could be deleted.